providers/implementations/keymgmt/rsa_kmgmt.c

```diff
@@ -157,8 +158,9 @@ static int key_to_params(RSA *rsa, OSSL_PARAM_BLD *tmpl)
 static void *rsa_importkey(void *provctx, const OSSL_PARAM params[])
 {
     RSA *rsa;
+    OPENSSL_CTX *libctx = PROV_LIBRARY_CONTEXT_OF(provctx);
 
-    if ((rsa = RSA_new()) == NULL
+    if ((rsa = rsa_new_with_ctx(libctx)) == NULL
```

Comment:
Here be dragons. I would much rather have a keymgmt structure that contains an OPENSSL_CTX and an RSA than embedding the former in the latter...